src/main/java/io/cryostat/net/web/http/api/v2/TargetProbesGetHandler.java

```diff
@@ -112,16 +116,25 @@ public IntermediateResponse<String> handle(RequestParameters requestParams) thro
                 getConnectionDescriptorFromParams(requestParams),
                 connection -> {
                     connection.connect();
+                    List<String> response = new ArrayList<String>();
                     AgentJMXHelper helper = new AgentJMXHelper(connection.getHandle());
                     String probes = helper.retrieveEventProbes();
+                    if (probes != null && !probes.isBlank()) {
+                        ProbeTemplate template = new ProbeTemplate();
+                        template.deserialize(new ByteArrayInputStream(probes.getBytes()));
+                        for (Event e : template.getEvents()) {
+                            response.add(e.toString());
```

Comment:
Can this handler return a `List<Event>` and allow the JSON serialization of the response (via Gson) to perform the output formatting, rather than calling `toString()` directly on each element here?

Reply:
Fixed this to use Gson to serialize.